app/Helpers/GetPushNotificationToSupporter.php

```diff
@@ -0,0 +1,208 @@
+<?php
+
+namespace App\Helpers;
+
+use App\Facades\PushNotification;
+use stdClass;
+use Throwable;
+use App\Models\Camp;
+use App\Models\User;
+use App\Models\Topic;
+use App\Models\Thread;
+use App\Models\Support;
+use App\Models\Nickname;
+
+class GetPushNotificationToSupporter
+{
+
+    public function pushNotificationToSupporter($request ,$topicNum, $campNum, $action = 'add', $threadId = null)
+    {
+        $directSupporter = Support::getDirectSupporter($topicNum, $campNum);
+        $subscribers = Camp::getCampSubscribers($topicNum, $campNum);
+        $directSupporterUser = [];
+        $directSupporterUserId = [];
+        $subscribers_user = [];
+        $subscribers_user_id = [];
+        foreach ($directSupporter as $supporter) {
+            $user = Nickname::getUserByNickName($supporter->nick_name_id);
+            $user_id = $user->id ?? null;
+            $directSupporterUser[] = $user;
+            $directSupporterUserId[] = $user_id;
+        }
+        if ($subscribers && count($subscribers) > 0) {
+            foreach ($subscribers as $sub) {
+                if (!in_array($sub, $directSupporterUserId, true)) {
+                    $userSub = User::find($sub);
+                    $subscribers_user[] = $userSub;
+                    $subscribers_user_id[] = $userSub->id;
+                }
+            }
+        }
+        $filtered_direct_supporter_user = array_unique($directSupporterUser);
+        $filtered_subscribers_user = array_unique($subscribers_user);
+        $topic = Topic::getLiveTopic($topicNum, "");
+        $filter['topicNum'] = $topicNum;
+        $filter['asOf'] = "";
+        $filter['campNum'] = $campNum;
+        $camp = Camp::getLiveCamp($filter);
+        $liveThread = Thread::find($threadId);
+        $PushNotificationData =  new stdClass();
+        $PushNotificationData->topic_num = $topic->topic_num;
+        $PushNotificationData->camp_num = $camp->camp_num;
+        if (isset($filtered_direct_supporter_user) && count($filtered_direct_supporter_user) > 0) {
+
+            foreach ($filtered_direct_supporter_user as $user) {
+                try {
+                    $PushNotificationData->user_id = $user->id;
+                    if ($action == 'add') {
```

Comment:
Can we refactor this nested if else {} also ?

Reply:
@akuks We will refactor it in some other branch. Please merge this branch. Because there are more issues in this branch.